packages/venia-concept/src/ServiceWorker/Utilities/messageHandler.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * handlers is an object that holds all the message
+ * handlers for the client messages. Key is the
+ * type of the message and value is an array of handlers.
+ *
+ * When a message arrives from the client, the
+ * message's type will be used to get the list of handlers
+ * that need to be invoked from the handlers object.
+ */
+const handlers = {};
```

Comment:
2 components listening for updates like background sync or high res image available. There is no use case as of today but I figured it will be useful moving forward. Dint want to make signature changes in the future. Also, this is how `addEventListener` works so it is just a borrowed idea.